packages/x-bus/src/__tests__/x-bus.spec.ts

```diff
@@ -1,7 +1,191 @@
+      // Given a priority bus
+      const queue = new XPriorityQueue();
+      const bus = new XPriorityBus({ queue });
+      const emittedEvents: XEvent[] = [];
+      const pushEmittedEvent: AnyFunction = ({ event }) => emittedEvents.push(event);
+
+      // When multiple events are emitted at the same time
+      // Then they're added to the queue
+      // And cancel an ongoing flushing of the queue
+      let pendingFlush = bus['pendingFlush'];
+      expect(pendingFlush).toBeUndefined();
+      expect(queue.isEmpty()).toBe(true);
+
+      const emit1 = bus.emit('TestEvent1', undefined, { priority: 2 }).then(pushEmittedEvent);
+      pendingFlush = bus['pendingFlush'];
+      expect(pendingFlush).toBeDefined();
+      expect(queue.size()).toBe(1);
+
+      const emit2 = bus.emit('TestEvent2', undefined, { priority: 6 }).then(pushEmittedEvent);
+      expect(pendingFlush).not.toBe(bus['pendingFlush']);
+      pendingFlush = bus['pendingFlush'];
+      expect(queue.size()).toBe(2);
+
+      const emit3 = bus.emit('TestEvent3', undefined, { priority: 4 }).then(pushEmittedEvent);
+      expect(pendingFlush).not.toBe(bus['pendingFlush']);
+      pendingFlush = bus['pendingFlush'];
+      expect(queue.size()).toBe(3);
+
+      const emit4 = bus.emit('TestEvent4', undefined, { priority: 0 }).then(pushEmittedEvent);
+      expect(pendingFlush).not.toBe(bus['pendingFlush']);
+      expect(queue.size()).toBe(4);
+
+      // The events are batched and emitted in the right order
+      jest.runAllTimers();
+      await Promise.all([emit1, emit2, emit3, emit4]);
+      expect(emittedEvents).toEqual(['TestEvent4', 'TestEvent1', 'TestEvent3', 'TestEvent2']);
+      expect(queue.isEmpty()).toBe(true);
+    });
+  });
+
+  describe('on', () => {
+    it('allows to create subscriptions to an event', () => {
+      // Given a bus
+      const bus = new XPriorityBus();
+
+      // Then creates a subscription to an event
+      const testEventSubscriptionFn = jest.fn();
+      const anotherTestEventSubscriptionFn = jest.fn();
+      bus.on('TestEvent').subscribe(testEventSubscriptionFn);
+      const anotherTestEventSubscription = bus
+        .on('TestEvent')
+        .subscribe(anotherTestEventSubscriptionFn);
+
+      // When that event is emitted
+      bus.emit('TestEvent');
+      bus.emit('TestEvent');
+      jest.runAllTimers();
+
+      anotherTestEventSubscription.unsubscribe();
+      bus.emit('TestEvent');
+      jest.runAllTimers();
+
+      // Then the bus notify its subscriptions
+      expect(testEventSubscriptionFn).toHaveBeenCalledTimes(3);
+      expect(anotherTestEventSubscriptionFn).toHaveBeenCalledTimes(2);
+    });
+  });
+
+  describe('getEventPriority', () => {
+    it('can override the priority behavior', async () => {
+      // Given a priority bus
+      const bus = new XPriorityBus();
+
+      // When the priority assignment function is overriden
+      bus.getEventPriority = (event, metadata) => metadata.customPriority;
```

Comment:
I have mixed feelings about this. Lets wait for the other reviewer thoughts on this.

Reply:
I've changed the visibility of the function to protected and removed the test. We decided that it was a little messy and in case you want a custom getEventPriority function you can always create your own bus implementation or we can later in the future expose this functionality. Additionally, a new property has been created, `defaultEventPriority`, to set a default priority for any event that doesn't have a priority defined in its metadata nor its name matches any key of the priorities dictionary.